aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -35,6 +42,9 @@ pub struct ProviderConfig {
     connector: HttpConnector,
     sleep: Option<Arc<dyn AsyncSleep>>,
     region: Option<Region>,
+    parsed_profile: Arc<OnceCell<Result<ProfileSet, ProfileFileLoadError>>>,
+    profile_files: ProfileFiles,
+    profile_name_override: Option<Cow<'static, str>>,
```

Comment:
It'd be nice to have docs for these somewhere. These are private fields so they don't need to be doc comments but something like this would be nice:
```suggestion
    // An AWS profile created from `ProfileFiles` and a `profile_name`
    profile: Arc<OnceCell<Result<ProfileSet, ProfileFileLoadError>>>,
    // A list of [std::path::Path]s to profile files
    profile_files: ProfileFiles,
    // An override to use when constructing a `ProfileSet`
    profile_name_override: Option<Cow<'static, str>>,
```

Otherwise I'm going to get confused about what each part is for.